File: data/code_pairs/fortran/1931.f90
program main
  implicit none

  integer, parameter :: size = 100
  integer :: array(size), i

  call kernels(array, size)
  do i = 1, size
     if (array(i) /= i) then
        print *, "Kernels Test FAILED."
        stop
     endif
  end do
  print *, "Kernels Test PASSED."
end program main

subroutine kernels(array, n)
  integer, dimension(n) :: array
  integer :: n, i

  do i = 1, n
     array(i) = i
  end do
end subroutine kernels

File: data/code_pairs/cpp/1931.cpp
#include <iostream>
#include <cstdlib> // For std::abort()

const int size = 100;

void kernels(int* array, int n) {
    for (int i = 0; i < n; ++i) {
        array[i] = i + 1; // Adjusting for 0-based indexing in C++
    }
}

int main() {
    int array[size];

    kernels(array, size);
    for (int i = 0; i < size; ++i) {
        if (array[i] != i + 1) { // Test failed
            std::cerr << "Kernels Test FAILED." << std::endl;
            std::abort();
        }
    }
    std::cout << "Kernels Test PASSED." << std::endl;

    return 0;
}